### ISSUE TITLE
mathtext broken with cairo backend as of master

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#8797]
As of master, with the gtk3cairo backend
```
title("$\\sqrt{2}$")
```
fails with
```
Traceback (most recent call last):
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/backends/backend_gtk3cairo.py", line 45, in on_draw_event
    self._render_figure(w, h)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/backends/backend_gtk3cairo.py", line 34, in _render_figure
    self.figure.draw (self._renderer)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/artist.py", line 68, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/figure.py", line 1242, in draw
    renderer, self, artists, self.suppressComposite)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/artist.py", line 68, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/axes/_base.py", line 2385, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/artist.py", line 68, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/text.py", line 801, in draw
    ismath=ismath, mtext=mtext)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/backends/backend_cairo.py", line 272, in draw_text
    self._draw_mathtext(gc, x, y, s, prop, angle)
  File "/home/antony/src/extern/matplotlib/lib/matplotlib/backends/backend_cairo.py", line 319, in _draw_mathtext
    self.fontweights[fontProp.weight])
KeyError: 'regular'
```
Probably just needs adding an entry in fontweights.
Bisects back to d94a6a526fcae5e29381d66624b216e8c255a0c0
Probably the cairo backend should get some testing too... (right now it's 0% covered)

**Comment by anntzer** [comment]
Allowing myself to retag as release critical as it does completely break a previously only half-broken backend :-) and there is an easy fix available.  Feel free to untag if you disagree.

**Comment by tacaswell** [comment]
Fixed by #8936
